code/controllers/subsystems/fluids.dm

```diff
@@ -41,14 +39,36 @@ var/datum/controller/subsystem/fluids/SSfluids
 		active_fluids_copied_yet = FALSE
 		af_index = 1
 
-	var/flooded_a_neighbor // Not used, required by FLOOD_TURF_NEIGHBORS.
 	var/list/curr_sources = processing_sources
+	var/list/checked = list()
 	while (curr_sources.len)
 		var/turf/T = curr_sources[curr_sources.len]
 		curr_sources.len--
-
-		FLOOD_TURF_NEIGHBORS(T, FALSE)
-
+		var/flooded_a_neighbor
+		for(var/spread_dir in GLOB.cardinal)
+			UPDATE_FLUID_BLOCKED_DIRS(T)
```

Comment:
Why do this 4 times rather than once outside the loop?

Reply:
Might be passing it the wrong argument, it's meant to update the turf we're stepping to.